packages/perseus/src/widgets/plotter.tsx

```diff
@@ -14,53 +13,64 @@ import KhanColors from "../util/colors";
 import GraphUtils from "../util/graph-utils";
 import KhanMath from "../util/math";
 
-import type {WidgetExports} from "../types";
+import type {APIOptions, WidgetExports} from "../types";
 
 const {deepEq} = Util;
 
-const BAR = "bar",
-    LINE = "line",
-    PIC = "pic",
-    HISTOGRAM = "histogram",
-    DOTPLOT = "dotplot";
-
-const widgetPropTypes = {
-    type: PropTypes.oneOf([BAR, LINE, PIC, HISTOGRAM, DOTPLOT]),
-    labels: PropTypes.arrayOf(PropTypes.string),
-    categories: PropTypes.arrayOf(
-        PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-    ),
-
-    scaleY: PropTypes.number,
-    maxY: PropTypes.number,
-    snapsPerLine: PropTypes.number,
-
-    picSize: PropTypes.number,
-    pixBoxHeight: PropTypes.number,
-    picUrl: PropTypes.string,
-
-    plotDimensions: PropTypes.arrayOf(PropTypes.number),
-    labelInterval: PropTypes.number,
-    starting: PropTypes.arrayOf(PropTypes.number),
-    static: PropTypes.bool,
-} as const;
-
-class Plotter extends React.Component<any, any> {
+const BAR = "bar";
```

Comment:
I'm going through this PR top-to-bottom. We have this same set of values in the plotter-editor. I _think_ we probably want to define it in one place (here, I'd think).  I noticed that we also have the `PerseusPlotterWidgetOptions` type in `perseus-types.ts` defining this option as only a `string`. If you feel adventurous you could move this enum def right into `perseus-types.ts`.

Reply:
I put on a band-aide on this for now. I'd like to talk about: https://khanacademy.atlassian.net/browse/LC-1117